local-cli/runIOS/RunIOSInternals.js

```diff
@@ -0,0 +1,272 @@
+/**
+* Copyright (c) 2015-present, Facebook, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the BSD-style license found in the
+* LICENSE file in the root directory of this source tree. An additional grant
+* of patent rights can be found in the PATENTS file in the same directory.
+*/
+'use strict';
+
+const child_process = require('child_process');
+const fs = require('fs');
+const path = require('path');
+const findXcodeProject = require('./findXcodeProject');
+const parseIOSDevicesList = require('./parseIOSDevicesList');
+const findMatchingSimulator = require('./findMatchingSimulator');
+
+function RunIOSInternals(os_device,
+                         os_simulator,
+                         os_scheme_suffix,
+                         os_type,
+                         os_run_name,
+                         os_default_simulator) {
+  this.os_device = os_device;
+  this.os_simulator = os_simulator;
+  this.os_scheme_suffix = os_scheme_suffix;
+  this.os_type = os_type;
+  this.os_run_name = os_run_name;
+  this.os_default_simulator = os_default_simulator;
+}
+
+RunIOSInternals.prototype.getBuildPath = function(configuration = 'Debug', appName, isDevice) {
+  return `build/Build/Products/${configuration}-${isDevice ? this.os_device : this.os_simulator}/${appName}.app`;
+};
+
+RunIOSInternals.prototype.runIOS = function(argv, config, args) {
```

Comment:
Rather than having those methods on prototype, prefer to keep them as `functions` or even in separate files for better testability (where it makes sense).